aries_cloudagent/commands/provision.py

```diff
@@ -35,6 +39,14 @@ async def provision(settings: dict):
 
         root_profile, public_did = await wallet_config(context, provision=True)
 
+        # store mediator invite url if provided
+        mediation_invite = settings.get("mediation.invite", None)
+        if mediation_invite:
+            async with root_profile.session() as session:
+                await MediationInviteStore(session.context.inject(BaseStorage)).store(
+                    mediation_invite
+                )
```

Comment:
why do we need to store the mediation invitation? Each connection stores information about the invitation used to establish the connection. Should we instead check for a connection made with the provided invitation, if no connection that was created by that invitation can be found, create a new one? In this way, we can restart aca-py using the same configuration with the same mediation invitation and not have a new connection created with every restart.

Reply:
I see. Do you think we should store the connection id somewhere as a pointer once it is found ? Or is the search space small enough that we can search for a connection at each startup ?

Reply:
I think I disagree with you on this one -- the invitations are not stored in an easily searchable format. We would have to iterate through all connections and retrieve the invitation associated with each and compare it against the invitation presented on the command line. That would be N + 1 storage operations where N is the number of connections rather than 1 retrieve and 1 update at most for this method.

Reply:
I understand the approach now, I was thinking the invitation was part of the connection data structure and was easily filtered on. Thanks for the patience.